detection_rules/rule.py

```diff
@@ -1309,6 +1313,9 @@ class TOMLRule:
     path: Optional[Path] = None
     gh_pr: Any = field(hash=False, compare=False, default=None, repr=False)
 
+    def __post_init__(self):
```

Comment:
we should do this much earlier in the build chain (remember its not uncommon to instantiate The `xData` classes directly, which would miss this.

* create a method `validates_data` in `BaseRuleData` and decorate it with `Marshmallow.validates_schema`
* move `validate_version_revision` to that same class
* call from `validates_data`

```python
class BaseRuleData(MarshmallowDataclassMixin, StackCompatMixin):
    ...

    @validates_schema
    def validates_data(self, data, **kwargs):
        self.validate_revision_data()

    def validate_revision_data()
```

Reply:
I thought about this, but something seems wrong about relying on an environment variable. I'll have to put a little more thought into how we should do this.

Reply:
Just so that I understand, what is the reason to do it earlier in the chain? Would we expect customers to want to use these fields? Or are we expecting to not have all of our pre-built rules be toml files? Thanks!

Reply:
In general, catching errors as early is possible is good practice, but in this repo, if someone wants to use our code and not just the toml file, we will be able to still catch this error.